### PR TITLE
Proxmox inventory: Added snapshots fact

### PR DESCRIPTION
**Pull request by Thulium-Drake**
##### SUMMARY

Add a list of existing snapshots to the host facts.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

proxmox inventory

##### ADDITIONAL INFORMATION

ansible-inventory --host foreman-400
```paste below
{
    "proxmox_boot": "order=scsi0;ide2;net0",
    "proxmox_cores": 2,
    "proxmox_digest": "ccee3b9a6dd8c14bbd36a3e3b283140960636d23",
    "proxmox_ide2": {
        "disk_image": "data:iso/CentOS-8.2.2004-x86_64-minimal.iso",
        "media": "cdrom"
    },
    "proxmox_memory": 12288,
    "proxmox_name": "foreman-400",
    "proxmox_net0": {
        "bridge": "vmbr0",
        "virtio": "2A:DD:72:8D:31:DD"
    },
    "proxmox_node": "carrier",
    "proxmox_numa": 0,
    "proxmox_ostype": "l26",
    "proxmox_parent": "clean",
    "proxmox_scsi0": {
        "disk_image": "main:vm-115-disk-0",
        "size": "100G"
    },
    "proxmox_scsihw": "disk_image=virtio-scsi-pci",
    "proxmox_smbios1": "uuid=10f553bd-8144-4a8f-aa31-9d37cb0ec05f",
    "proxmox_snapshots": [
        "clean"
    ],
    "proxmox_sockets": 1,
    "proxmox_status": "stopped",
    "proxmox_vmgenid": "46baf711-59fc-452a-94b3-e88ed5ef82e4",
    "proxmox_vmid": "115",
    "proxmox_vmtype": "qemu",
}
```

### CODE REVIEW

File: changelogs/fragments/3044-proxmox-inventory-snapshots.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+new_features:
+  - proxmox inventory - added snapshots to host facts (https://github.com/ansible-collections/community.general/pull/3044)
```
Comment (Ajpantuso) [inline changelogs/fragments/3044-proxmox-inventory-snapshots.yml:2]:
```suggestion
minor_changes:
  - proxmox inventory plugin - added snapshots to host facts (https://github.com/ansible-collections/community.general/pull/3044).
```

File: plugins/inventory/proxmox.py [hunk 2]
```diff
@@ -307,6 +307,19 @@ def _get_vm_status(self, node, vmid, vmtype, name):
...
+        ret = self._get_json("%s/api2/json/nodes/%s/%s/%s/snapshot" % (self.proxmox_url, node, vmtype, vmid))
+
+        snapshots_key = 'snapshots'
+        snapshots_key = self.to_safe('%s%s' % (self.get_option('facts_prefix'), snapshots_key.lower()))
+
+        snapshots = []
+
+        for snapshot in ret:
+            if snapshot['name'] != 'current':
+                snapshots.append(snapshot['name'])
```
Comment (Ajpantuso) [inline plugins/inventory/proxmox.py:320]: How about the following?
```suggestion
        snapshots = [snapshot['name'] for snapshot in ret if snapshot['name'] != 'current']
```

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
Just a couple of suggestions, but functionality looks good.

File: changelogs/fragments/3044-proxmox-inventory-snapshots.yml (hunk 1 above)
Comment (Thulium-Drake) [reply changelogs/fragments/3044-proxmox-inventory-snapshots.yml]: Done!

File: plugins/inventory/proxmox.py (hunk 2 above)
Comment (Thulium-Drake) [reply plugins/inventory/proxmox.py]: Thanks for the tip! (I'm not a frequent programmer ;-) )

File: changelogs/fragments/3044-proxmox-inventory-snapshots.yml (hunk 1 above)
Comment (Ajpantuso) [reply changelogs/fragments/3044-proxmox-inventory-snapshots.yml]: Can you add a `.` to the end of the sentence as well? After that this is good to go, thanks!

File: changelogs/fragments/3044-proxmox-inventory-snapshots.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - proxmox inventory plugin - added snapshots to host facts. (https://github.com/ansible-collections/community.general/pull/3044)
```
Comment (felixfontein) [inline changelogs/fragments/3044-proxmox-inventory-snapshots.yml]:
```suggestion
  - proxmox inventory plugin - added snapshots to host facts (https://github.com/ansible-collections/community.general/pull/3044).
```

**Review by Ajpantuso — APPROVED** [review]
LGTM, thanks!

**Comment by felixfontein** [comment]
@Thulium-Drake thanks for implementing this!
@Ajpantuso thanks for reviewing!

**Pull request merged**